bottlerocket/agents/src/sonobuoy.rs

```diff
@@ -426,6 +527,18 @@ mod test_sonobuoy {
         assert_eq!(result.other_info.unwrap(), "e2e: {\"name\":\"e2e\",\"node\":\"global\",\"timestamp\":\"2022-12-08T15:37:23.007805243Z\",\"msg\":\"Test Suite completed\",\"total\":1,\"completed\":1}");
     }
 
+    #[test]
+    fn test_process_incomplete_results_progress_object() {
+        let result =
+            process_incomplete_sonobuoy_test_results(
+                &json!({"plugins":[{"plugin":"e2e","progress":{"name":"e2e","node":"global","timestamp":"2022-12-08T15:37:23.007805243Z","msg":"Test Suite completed","total":1,"completed":1},"status":"complete","result-status":"timed-out","result-counts":{"failed":1}}]})).unwrap();
+        assert_eq!(result.num_passed, 1);
+        assert_eq!(result.num_failed, 0);
+        assert_eq!(result.num_skipped, 0);
+        assert_eq!(result.outcome, Outcome::InProgress);
+        assert_eq!(result.other_info.unwrap(), "Running");
```

Comment:
I'm slightly confused about this unit test. The json there says the status is complete but we're asserting the result is `Outcome::InProgess`.

Reply:
It's because it is using the incomplete status results.

Reply:
`"result-status":"timed-out"`
Is it because the `result-status` isn't complete? If we're going by the codepath, `wait_for_sonobuoy_results` returns when the `status` is no longer `running` right so this case should theoretically never happen?

Reply:
Correct that code path shouldn't happen, I can delete the test if you want.

Reply:
Yes, I think we should delete this unit test if we don't expect this to ever happen. The presence of this unit test makes me think this is a possibility we want to test for (a very weird one at that).